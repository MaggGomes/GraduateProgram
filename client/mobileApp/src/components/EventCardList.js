import React, { Component } from 'react';
import { FlatList, View, Text, StyleSheet, TouchableOpacity, ImageBackground } from 'react-native';

/* Horizontal list presenting each day of an event */
export default class EventCardList extends Component {

    _renderItem = ({item}) => (
        <TouchableOpacity style={styles.card}
            onPress={() => {
                this.props.navigation.navigate('Day', {
                    id: item._id,
                    title: this.props.name+'_'+item.name
                });
            }}
        >
            <ImageBackground style={styles.cardBackground} source={{uri: item.image}}>
                <View style={styles.cardFooter}>
                    <Text style={styles.textFooter}>{item.name}</Text>
                </View>
            </ImageBackground>
        </TouchableOpacity>
    );

    _keyExtractor = (item) => item._id;

    render() {
        return (
            <View style={styles.container}>
                <FlatList
                    horizontal={true}
                    showsHorizontalScrollIndicator={false}
                    data={this.props.days}
                    renderItem={this._renderItem}
                    keyExtractor={this._keyExtractor}
                />
            </View>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        width: '100%',
        height: 175,
        paddingLeft:20,
        paddingRight: 20
    },
    card: {
        marginRight:15
    },
    cardBackground: {
        width: 140,
        height: 175,
        flexDirection: 'row',
        alignItems: 'flex-end',

    },
    cardFooter: {
        width: '100%',
        height: 40,
        backgroundColor: 'rgba(191,191,191,.8)',
        alignItems: 'center',
        justifyContent: 'center'
    },
    textFooter: {
        color: 'white',
        fontSize: 15,
        textAlign: 'center'
    }
});

