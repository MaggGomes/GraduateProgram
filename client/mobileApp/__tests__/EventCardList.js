import React from 'react';
import renderer from 'react-test-renderer';
import EventCardList from '../src/components/EventCardList';

describe('EventCardList component', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<EventCardList />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});