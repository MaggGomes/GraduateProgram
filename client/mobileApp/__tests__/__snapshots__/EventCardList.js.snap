// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventCardList component renders correctly 1`] = `
<View
  style={
    Object {
      "height": 175,
      "paddingLeft": 20,
      "paddingRight": 20,
      "width": "100%",
    }
  }
>
  <RCTScrollView
    data={undefined}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    initialNumToRender={10}
    invertStickyHeaders={undefined}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
